### PR TITLE
Add `content-type` header for JSON responses

### PR DESCRIPTION
**Pull request by adieu**
I'm running `deck` behind a reverse proxy which would enable MIME type checking with `x-content-type-options: nosniff` header.

`deck` would always show empty page because the browser blocked all jsonp requests as its `content-type` is always 'text/plain`.

This PR set proper `content-type` for JSON responses to fix the issue.

### CODE REVIEW

**Comment by matthyx** [comment]
Is `deck` the only component with that issue? Or do you see other parts that would benefit from the `content-type` header?

**Comment by adieu** [comment]
`deck` is the only component serving http requests in my setup. Not quite sure if there're other components providing web interface or not.

**Comment by matthyx** [comment]
`prstatus` if I remember correctly?

I mean maybe you could have a quick look in the `test-infra` repo, it wouldn't hurt to fix all components at the same time...

**Comment by adieu** [comment]
Sure.

**Comment by adieu** [comment]
I've run a grep on `prow`. It seems `deck` is the only user facing component.

```
$ grep -r '\.Handle(' prow
./cmd/deck/main.go:     mux.Handle("/static/", http.StripPrefix("/static", staticHandlerFromDir(o.staticFilesLocation)))
./cmd/deck/main.go:     mux.Handle("/config", gziphandler.GzipHandler(handleConfig(cfg)))
./cmd/deck/main.go:     mux.Handle("/favicon.ico", gziphandler.GzipHandler(handleFavicon(o.staticFilesLocation, cfg)))
./cmd/deck/main.go:     mux.Handle("/pr", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "pr.html", nil)))
./cmd/deck/main.go:     mux.Handle("/command-help", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "command-help.html", nil)))
./cmd/deck/main.go:     mux.Handle("/plugin-help", http.RedirectHandler("/command-help", http.StatusMovedPermanently))
./cmd/deck/main.go:     mux.Handle("/tide", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "tide.html", nil)))
./cmd/deck/main.go:     mux.Handle("/tide-history", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "tide-history.html", nil)))
./cmd/deck/main.go:     mux.Handle("/plugins", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "plugins.html", nil)))
./cmd/deck/main.go:     mux.Handle("/github-login", gziphandler.GzipHandler(handleSimpleTemplate(o, cfg, "github-login.html", nil)))
./cmd/deck/main.go:     mux.Handle("/data.js", gziphandler.GzipHandler(handleData(ja)))
./cmd/deck/main.go:     mux.Handle("/prowjobs.js", gziphandler.GzipHandler(handleProwJobs(ja)))
./cmd/deck/main.go:     mux.Handle("/badge.svg", gziphandler.GzipHandler(handleBadge(ja)))
./cmd/deck/main.go:     mux.Handle("/log", gziphandler.GzipHandler(handleLog(ja)))
./cmd/deck/main.go:     mux.Handle("/rerun", gziphandler.GzipHandler(handleRerun(prowJobClient)))
./cmd/deck/main.go:             mux.Handle("/plugin-help.js",
./cmd/deck/main.go:             mux.Handle("/tide.js", gziphandler.GzipHandler(handleTidePools(cfg, ta)))
./cmd/deck/main.go:             mux.Handle("/tide-history.js", gziphandler.GzipHandler(handleTideHistory(ta)))
./cmd/deck/main.go:             mux.Handle("/pr-data.js", handleNotCached(
./cmd/deck/main.go:             mux.Handle("/github-login", goa.HandleLogin(oauthClient))
./cmd/deck/main.go:             mux.Handle("/github-login/redirect", goa.HandleRedirect(oauthClient, githuboauth.NewGitHubClientGetter()))
./cmd/deck/main.go:             redirectMux.Handle("/", func(oldMux *http.ServeMux, host string) http.HandlerFunc {
./cmd/deck/main.go:     mux.Handle("/spyglass/static/", http.StripPrefix("/spyglass/static", staticHandlerFromDir(o.spyglassFilesLocation)))
./cmd/deck/main.go:     mux.Handle("/spyglass/lens/", gziphandler.GzipHandler(http.StripPrefix("/spyglass/lens/", handleArtifactView(o, sg, cfg))))
./cmd/deck/main.go:     mux.Handle("/view/", gziphandler.GzipHandler(handleRequestJobViews(sg, cfg, o)))
./cmd/deck/main.go:     mux.Handle("/job-history/", gziphandler.GzipHandler(handleJobHistory(o, cfg, c)))
./cmd/deck/main.go:     mux.Handle("/pr-history/", gziphandler.GzipHandler(handlePRHistory(o, cfg, c)))
./cmd/hook/main.go:     http.Handle("/hook", server)
./cmd/hook/main.go:     http.Handle("/plugin-help", pluginhelp.NewHelpAgent(pluginAgent, githubClient))
./cmd/jenkins-operator/main.go: http.Handle("/", gziphandler.GzipHandler(handleLog(jc)))
./cmd/sub/main.go:      http.Handle("/push", pushServer)
./cmd/tide/main.go:     http.Handle("/", c)
./cmd/tide/main.go:     http.Handle("/history", c.History)
./external-plugins/cherrypicker/main.go:        http.Handle("/", server)
./external-plugins/needs-rebase/main.go:        http.Handle("/", server)
./external-plugins/refresh/main.go:     http.Handle("/", serv)
./metrics/metrics.go:   metricsMux.Handle("/metrics", promhttp.Handler())
```

**Comment by matthyx** [comment]
Thanks!
/lgtm

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -603,8 +611,10 @@ func handleProwJobs(ja *jobs.JobAgent) http.HandlerFunc {
 		// If we have a "var" query, then write out "var value = {...};".
 		// Otherwise, just write out the JSON.
 		if v := r.URL.Query().Get("var"); v != "" {
```
Comment (stevekuznetsov) [inline prow/cmd/deck/main.go]: Could we DRY this bit out as well while we're here? A helper that checks the query and writes the appropriate thing

Comment (adieu) [reply prow/cmd/deck/main.go]: Sounds good to me. I'll improve the patch.

**Pull request merged**